azure/template.json

```diff
@@ -284,7 +291,20 @@
     "databaseResourceNamePrefix": "[toLower(concat(parameters('resourceEnvironmentName'),'-', parameters('databaseServiceName')))]",
     "databaseServerName": "[concat(variables('databaseResourceNamePrefix'), '-psql')]",
     "databaseUsername": "[concat(parameters('databaseUsername'), '@', variables('databaseServerName'))]",
-    "storageAccountName": "[replace(concat(variables('resourceNamePrefix'), 'str'), '-', '')]"
+    "storageAccountName": "[replace(concat(variables('resourceNamePrefix'), 'str'), '-', '')]",
+    "defaultAvailabilityCheckHosts": "[if(greater(length(parameters('customHostName')), 0), createArray(concat('azcheck:', variables('appServiceName'), '.azurewebsites.net/ping'), concat('check:', parameters('customHostName'), '/ping')), createArray(concat('azcheck:', variables('appServiceName'), '.azurewebsites.net/ping')))]",
+    "availabilityCheckHosts": "[if(greater(length(parameters('customAvailabilityMonitors')), 0), concat(variables('defaultAvailabilityCheckHosts'), parameters('customAvailabilityMonitors')), variables('defaultAvailabilityCheckHosts'))]",
```

Comment:
```C
if(greater(length(parameters('customAvailabilityMonitors')), 0),
   concat(
     variables('defaultAvailabilityCheckHosts'),
     parameters('customAvailabilityMonitors')
   ),
   variables('defaultAvailabilityCheckHosts')
)
```

Do these not need the `azcheck:` prefixes?

Reply:
No. `azcheck` is inside defaultAvailabilityCheckHosts variable.  Here we are just checking if custom monitors are provided as a parameter and taking actions accordingly.